content-src/components/LinkMenu/LinkMenu.js

```diff
@@ -38,9 +38,16 @@ const LinkMenu = React.createClass({
           label: "Dismiss",
           icon: "dismiss",
           userEvent: "BLOCK",
-          onClick: () => dispatch(actions.NotifyBlockURL(site.url))
+          onClick: () => {
+            if (site.recommended) {
+              dispatch(actions.NotifyBlockRecommendation(site.url));
+              dispatch(actions.RequestHighlightsLinks());
```

Comment:
If this needs to be re-requested, you might run into a race condition situation here. Maybe trigger re-firing highlights from the add-on side, when the pref changes?